796.py

```diff
@@ -1,18 +1,3 @@
 class Solution:
     def rotateString(self, A: str, B: str) -> bool:
-        if len(A) != len(B):
-            return False
-        
-        if len(A) == 0 and len(B) == 0:
-            return True
-        
-        length = len(A)
-        
-        while length > 0:
-            length -= 1
-            A = A[1:len(A)] + A[0:1]
-            
-            if A == B:
-                return True
-        
-        return False
+        return len(A) == len(B) and B in A + A
```

Comment:
LGTM. Checking whether or not `B` is in `A + A` only takes linear time `O(length)`. In your previous implementation, you checked `A == B` for `length` times, and each check takes `O(length)` time, making the overall complexity `O(length^2)`, which was not good.